governator-core/src/main/java/com/netflix/governator/event/ApplicationEventModule.java

```diff
@@ -0,0 +1,115 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import com.google.inject.matcher.Matchers;
+import com.google.inject.spi.InjectionListener;
+import com.google.inject.spi.TypeEncounter;
+import com.google.inject.spi.TypeListener;
+
+/**
+ * Adds support for passing {@link ApplicationEvent}s backed by Guava {@link EventBus}
+ * See {@link EventListener} and {@link ApplicationEventPublisher} for usage. 
+ */
+public class ApplicationEventModule extends AbstractModule {
+    
+    private static final Logger LOG = LoggerFactory.getLogger(ApplicationEventModule.class);
+    private final EventBus eventBus = new EventBus("Governator Application EventBus");
+    private final ApplicationEventSubscribingTypeListener subscribingTypeListener = new ApplicationEventSubscribingTypeListener(eventBus);
+    
+
+    @Singleton
```

Comment:
Can we move this out to a Guava specific module so we decouple the Guava implementation from the API and TypeListener